unstructured/partition/pdf.py

```diff
@@ -226,8 +229,12 @@ def _partition_pdf_or_image_local(
             ocr_languages=ocr_languages,
             extract_tables=infer_table_structure,
         )
+    elements = document_to_element_list(layout, include_page_breaks=include_page_breaks)
+    for el in elements:
+        if isinstance(el, Text):
+            el.text = re.sub(RE_MULTISPACE, " ", el.text.replace("\n", " ")).strip()
 
-    return document_to_element_list(layout, include_page_breaks=include_page_breaks)
+    return [el for el in elements if not isinstance(el, Text) or el.text]
```

Comment:
can this code be rewritten to not require staring it a bit to understand the intent?

since we're already looping through `elements` in line 233 anyway, why not just append to `return_elements` there and then the logic of what is getting returned should be nice and clear?

Reply:
In general, if we're dropping anything without text, won't this also remove all page breaks being injected by `document_to_element_list`?

Reply:
ah, not if `text` is not empty, which i think it wouldn't be if it is `"<Page Break>"` (which it probably shouldn't be in the first place).

i think this will also be clearer to discuss if the logic is in the loop and not list comprehension to begin with.

CC @qued , soliciting comments on this thread and my other comment

Reply:
Ah didn't realize the text of the PageBreak element was itself `<PAGE BREAK>`. I looked at the constructor and saw ```
text: Optional[str] = None,
```
Thinking there would be no text associated with that element by default.